core/src/main/java/org/jdbi/v3/core/mapper/BuiltInMapperFactory.java

```diff
@@ -87,6 +87,8 @@
         mappers.put(OffsetDateTime.class, referenceMapper(BuiltInMapperFactory::getOffsetDateTime));
         mappers.put(ZonedDateTime.class, referenceMapper(BuiltInMapperFactory::getZonedDateTime));
         mappers.put(LocalTime.class, referenceMapper(BuiltInMapperFactory::getLocalTime));
+
+        mappers.put(Object.class, referenceMapper(ResultSet::getObject));
```

Comment:
I'm worried that this could cause bizarre `ClassCastException`s... the driver might guess a different type for `getObject` than we do? I'd prefer to not change it unless we understand why.